jdaviz/configs/specviz/plugins/parsers.py

```diff
@@ -76,10 +78,12 @@ def specviz_spectrum1d_parser(app, data, data_label=None, format=None, show_in_v
 
     # If there's already data in the viewer, convert units if needed
     current_unit = None
-    current_spec = app.get_data_from_viewer("spectrum-viewer")
-    if current_spec != {} and current_spec is not None:
-        spec_key = list(current_spec.keys())[0]
-        current_unit = current_spec[spec_key].spectral_axis.unit
+    if spectrum_viewer_reference_name in app.get_viewer_reference_names():
+        current_spec = app.get_data_from_viewer(spectrum_viewer_reference_name)
+        if current_spec != {} and current_spec is not None:
+            spec_key = list(current_spec.keys())[0]
+            current_unit = current_spec[spec_key].spectral_axis.unit
+
```

Comment:
Similar to https://github.com/spacetelescope/jdaviz/pull/1681#discussion_r990247038, I don't see this same check in the specviz2d helper. Struggling to see why this is only needed for specviz?